render/gles2/renderer.c

```diff
@@ -182,11 +191,14 @@ static void gles2_render_quad_with_matrix(struct wlr_renderer *wlr_renderer,
 	float transposition[9];
 	wlr_matrix_transpose(transposition, matrix);
 
+	float converted[4];
+	color_convert(NULL, wlr_renderer->color, color, converted);
```

Comment:
Of course this won't even do anything without a protocol to provide a client profile or colorspace. Which is why this _really_ needs a protocol before it can be merged. Otherwise people who currently care about color and let clients apply the profile themselves will be broken as they'll have a profile applied twice before display.

Reply:
This case is covered - for color managed clients you can specify a color profile that matches the monitor.

Reply:
And how exactly is that sent such that the compositor doesn't do it again?

Reply:
Take a look at the sway PR https://github.com/swaywm/sway/pull/5586 - the ICC profile is set for each window.

Reply:
hm neither this nor the sway PR seem to [register with colord](https://gitlab.freedesktop.org/wayland/weston/-/blob/master/compositor/cms-colord.c) which is what would *inform* the clients of the profile so that they could do the conversion themselves

Reply:
Correct - this is just basic manual color management - you specify the profiles for windows and outputs in your sway config.

Reply:
colord is out-of-scope for wlroots.

Reply:
I don't understand you. If you want, you can completely ignore it - things will work as they do now. Otherwise, it's the backend solution which can be built on with a protocol, in which case the config will just provide defaults.

Reply:
What cyanreg is talking about is this https://gitlab.freedesktop.org/wayland/wayland-protocols/-/merge_requests/14. There is a WIP upstream protocol for color management. Right now it doesn't have HDR but you could help with that. I think the benefit of standardizing this with a protocol is that you could start getting patches for client support. Wayland protocols also need an implementation before they're merged, so if you modified your patches to work with this protocol that would count.

Reply:
Support for that protocol would better fit in a different PR, since the concerns are mostly orthogonal.
- This PR is assumes a `wlr_color_config` (being a profile or something else) is assigned to each `wlr_surface` or output (`wlr_renderer`) and applies them during rendering.
- The protocol is about reading and writing the color config assign to surfaces and outputs via clients, so it's the other side of the equation.